frontend/src/views/Collection/utils.tsx

```diff
@@ -2,15 +2,39 @@ import { Classes } from "@blueprintjs/core";
 import React from "react";
 import {
   Collection,
+  COLLECTION_LINK_TYPE,
   COLLECTION_LINK_TYPE_OPTIONS,
   DATASET_ASSET_FORMAT,
   Link,
 } from "src/common/entities";
 import { getUrlHost } from "src/common/utils/getUrlHost";
 import { StyledLink } from "./common/style";
 
+const LINK_ORDER: COLLECTION_LINK_TYPE[] = [
+  COLLECTION_LINK_TYPE.DOI,
+  COLLECTION_LINK_TYPE.DATA_SOURCE,
+  COLLECTION_LINK_TYPE.RAW_DATA,
+  COLLECTION_LINK_TYPE.PROTOCOL,
+  COLLECTION_LINK_TYPE.LAB_WEBSITE,
+  COLLECTION_LINK_TYPE.OTHER,
+];
+
 export function renderLinks(links: Link[]) {
-  return links?.map(({ link_url: url, link_type: type }) => {
+  const linkTypesToLinks = createLinkTypesToLinks(links);
```

Comment:
this creates a map of link type as key to the link type's links as value, so we can use the map to create `orderedLinks` based on `LINK_ORDER`, cutting down runtime from O(MN) to O(M + N)